src/Util/ClassSourceManipulator.php

```diff
@@ -121,13 +121,23 @@ public function addEntityField(ClassProperty $mapping): void
 
         $defaultValue = null;
         $commentLines = [];
-        if ('array' === $typeHint && !$nullable) {
-            $defaultValue = new Node\Expr\Array_([], ['kind' => Node\Expr\Array_::KIND_SHORT]);
-            if (null !== $mapping->enumType) {
+
+        if (null !== $mapping->enumType) {
+            if ('array' === $typeHint) {
+                // still need to add the use statement
+                $this->addUseStatementIfNecessary($mapping->enumType);
+
                 $commentLines = [\sprintf('@return %s[]', Str::getShortClassName($mapping->enumType))];
+                if ($nullable) {
+                    $commentLines[0] .= '|null';
+                } else {
+                    $defaultValue = new Node\Expr\Array_([], ['kind' => Node\Expr\Array_::KIND_SHORT]);
+                }
+            } else {
+                $typeHint = $this->addUseStatementIfNecessary($mapping->enumType);
             }
```

Comment:
For clarity to my suggestion above, removing that use statement call and moving this one outside of the `if` conditional would accomplish the same thing with less code if I'm not mistaken.
```suggestion
            }
            
            $typeHint = $this->addUseStatementIfNecessary($mapping->enumType);
```

Reply:
This will override `$typeHint` with the enum type even if we use an array, which was the initial problem. 
We don't want to use the returned type of `addUseStatementIfNecessary` if we specify an array.